### PR TITLE
Simplify some of the RPMMonitor logic

### PR DESCRIPTION
**Pull request by penguinland**
Rand and I are struggling with https://viam.atlassian.net/browse/RSDK-694, and although this is not a fix, it is some refactoring which I think makes it easier to reason about this code (splitting the code into smaller, more self-contained single-purpose functions, adding comments, simplifying the control flow). I suggest reviewing commit-by-commit, because everything together gets muddled. For commits where a large section of code has changed indentation amounts, append a `?w=1` to the end of the URL to ignore whitespace changes (e.g., https://github.com/viamrobotics/rdk/commit/b8cf4502dc8caa5af2f66de33fae2a5050017c84?w=1 instead of https://github.com/viamrobotics/rdk/commit/b8cf4502dc8caa5af2f66de33fae2a5050017c84). 

For the most part, changes to behavior are not intended, though the mention of `state.regulated` in `goForInternal()` feels like it should fix a bug when the motor receives many different commands nearly simultaneously. It assumes that the comment I added is correct; if the comment is wrong, that change to the behavior might be wrong, too.

### CODE REVIEW

**Comment by edaniels** [comment]
Added Eliot since he's authored a lot of this

File: components/motor/gpio/motor_encoder.go [hunk 1]
```diff
@@ -471,8 +484,10 @@ func (m *EncodedMotor) goForInternal(ctx context.Context, rpm, revolutions float
 	defer m.stateMu.Unlock()
 
 	if revolutions == 0 {
+		// Moving 0 revolutions is a special value meaning "move forever."
```
Comment (erh) [inline components/motor/gpio/motor_encoder.go:487]: this comment is correct, but i think the change making regulated true is wrong.
regulated means we're going to a position.
which is the opposite of the comment

File: components/motor/gpio/motor_encoder.go [hunk 2]
```diff
@@ -372,56 +374,67 @@ func (m *EncodedMotor) rpmMonitorPass(pos, lastPos, now, lastTime int64, rpmDebu
...
-		} else {
-			neededPowerPct = math.Min(neededPowerPct, -0.01)
-			neededPowerPct = math.Max(neededPowerPct, -1)
-		}
+	// The last power percent resulted in the last RPM measurement. To get to the desired RPM,
+	// multiply by their ratio.
+	neededPowerPct := lastPowerPct * dOverC
 
-		newPowerPct = m.computeRamp(lastPowerPct, neededPowerPct)
+	if !math.Signbit(neededPowerPct) { // neededPowerPct is positive
```
Comment (randhid) [inline components/motor/gpio/motor_encoder.go:411]: Add comment above: 	// Bound the value of the power between 0.01 or 1 in the positive or negative direction.

**Review by randhid — APPROVED** [review]
one nit in 411 of `motor_encoder.go`. I think it's a good simplification.

**Comment by penguinland** [comment]
Changes made! Thank you both for the feedback. Eliot, I had misunderstood what `regulated` meant, and thanks for setting me straight.

**Pull request merged**